src/qgitrepository.cpp

```diff
@@ -624,4 +627,19 @@ bool Repository::shouldIgnore(const QString &path) const
     return result;
 }
 
+void Repository::setIdentity(const Identity &id)
+{
+    const auto name = !id.name.isEmpty() ? id.name.toUtf8() : nullptr;
+    const auto email = !id.email.isEmpty() ? id.email.toUtf8() : nullptr;
```

Comment:
I couldn't without doubt think what the compiler thinks `auto` will become. I thought it could become a `char*`, and (when the `QString` is not empty) a pointer to a data of a temporary `QByteArray` (returned by the `toUtf8()` call) would be taken.

Turned out that's not the case and `auto` will actually become `QByteArray`. To avoid this confusion, perhaps replacing the `nullptr`s with `QByteArray()` would help the reader.

Reply:
Should be clearer now.